### ISSUE TITLE
Unable to focus balloon toolbars (image toolbar, table toolbar etc)

### ISSUE THREAD

**Issue by jodator** [ckeditor/ckeditor5#5146]
As selected here (sorry for image with text) I'm not able to move focus to a visible baloon toolbar with keyboard.

![selection_011](https://user-images.githubusercontent.com/247363/38980532-5a7d243c-43bd-11e8-884f-edf2798c5b7d.png)

I've tried:
- ALT+F10 - obviously moves to main toolbar
- TAB - moves to image caption filed

Same is true for other toolbar like table, media embed etc.

---

If you'd like to see this feature implemented, add 👍 to this post.

**Comment by Reinmar** [comment]
The idea here would be that Alt+F10 could cycle through all possible toolbars (imaga/table/widget, balloon, classic, block, and so on...).

**Comment by fhalna** [comment]
Hello everyone,
Many thanks for the work done for the accessibility of Ckeditor 5, unfortunately this bug related to the possibility of inserting alternatives to images in the editor makes the editor completely inaccessible to blind users.

is it possible to change the milestone of this ticket?

**Comment by sunesimonsen** [comment]
It would be nice with a general fix that allows navigating to a contextual toolbar registered with `WidgetToolbarRepository`. 

I would expect `Tab` to focus the first item and escape to move focus back to the selected editor element.

**Comment by wimleers** [comment]
FYI: @bramd re-discovered this bug using NVDA at Drupal Dev Days Ghent 2022 — see https://wimleers.com/blog/ddd-ghent-2022 & https://twitter.com/gaborhojtsy/status/1512059083129593858.



> The idea here would be that Alt+F10 could cycle through all possible toolbars (imaga/table/widget, balloon, classic, block, and so on...).

This sounds relatively simple to implement I think? If you can give us guidance, maybe we could help make this happen, to address this hard blocker ASAP. 🤞 

But it is also _not_ what https://ckeditor.com/docs/ckeditor5/latest/features/keyboard-support.html#user-interface-and-navigation says:

> Move focus to the visible contextual balloon: `Tab`

So … what is the "visible contextual balloon" actually referring to? 🤔

**Comment by mlewand** [comment]
First of all let's make this issue a generic issue for any dynamically added/context toolbar (image toolbar, table toolbar, media embed toolbar etc).

One idea as mentioned is to make one hotkey cycle though all toolbars (good benefit is the discoverability of the feature).

However, let's see if there are any applications out there that have this problem solved so that we can have interoperability with it.

**Comment by wimleers** [comment]
Sounds good! 👍 

FYI: tracking this at https://www.drupal.org/project/drupal/issues/3283800 😊

**Comment by mlewand** [comment]
#3184 marked as a (partial) duplicate of this task.

**Comment by tony1377** [comment]
Any idea roughly when this might make it to release?

**Comment by Reinmar** [comment]
99% next release (end of August). See ckeditor/ckeditor5#12064
